URA-859 Assess data from validation runs - BunnyNo

Arun-bioinformatics opened the pull request:
Init repository

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16)
<!-- Reviewable:end -->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Added a new task entry for assessing data from validation runs related to the TE-97630453 panel design in the documentation.
  - Introduced new functions for processing and analysing VCF files, including filtering, transforming, and visualising genomic data.

- **Bug Fixes**
  - Ensured correct formatting and inclusion of links in the documentation for existing and new tasks.

- **Documentation**
  - Updated the README to reflect the new task and maintain clarity in ongoing work.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
URA-859 Assess data from validation runs

Init repository

pep8speaks commented:
Hello @Arun-bioinformatics! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`URA-859/coverage_plots.py`](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/coverage_plots.py):

> [Line 49:44](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/coverage_plots.py#L49): [W605](https://duckduckgo.com/?q=pep8%20W605) invalid escape sequence '\.'

* In the file [`URA-859/diagnostic_venn.py`](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/diagnostic_venn.py):

> [Line 63:48](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/diagnostic_venn.py#L63): [W605](https://duckduckgo.com/?q=pep8%20W605) invalid escape sequence '\.'
> [Line 63:54](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/diagnostic_venn.py#L63): [W605](https://duckduckgo.com/?q=pep8%20W605) invalid escape sequence '\.'
> [Line 64:48](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/diagnostic_venn.py#L64): [W605](https://duckduckgo.com/?q=pep8%20W605) invalid escape sequence '\.'
> [Line 64:54](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/diagnostic_venn.py#L64): [W605](https://duckduckgo.com/?q=pep8%20W605) invalid escape sequence '\.'
> [Line 198:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/diagnostic_venn.py#L198): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)

* In the file [`URA-859/plots_bcftools.py`](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py):

> [Line 51:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L51): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 75:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L75): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 112:44](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L112): [W605](https://duckduckgo.com/?q=pep8%20W605) invalid escape sequence '\.'
> [Line 112:50](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L112): [W605](https://duckduckgo.com/?q=pep8%20W605) invalid escape sequence '\.'
> [Line 289:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L289): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 375:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L375): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 482:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L482): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 483:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L483): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 491:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L491): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 492:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L492): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 502:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L502): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 505:80](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L505): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)



##### Comment last updated at 2025-01-02 14:25:34 UTC

Commit:
include_coverage_code

coverage_plots.py added.

Commit:
Coderabbit AIs suggestions

Delete lines 557 and 570

Commit:
Included scripts used for testing diagnostic samples.

nadia31415 reviewed (changes requested):
The PEP-8 issues mentioned should be addressed 

> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16#-:-OFgC5Ds4PS0G9WNC0DF:baw3y3w)* status: all files reviewed, 4 unresolved discussions (waiting on @Arun-bioinformatics)

___
*[`URA-859/diagnostic_venn.py` line 96 at r3](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16#-OFvmqsq05mDkQSLzK08:-OFvmqsq05mDkQSLzK09:b-hqzukb) ([raw file](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/diagnostic_venn.py#L96)):*
> ```Python
>     output_dir = os.path.join(output_folder)
> 
>     # Run bcftools isec
> ```

More for optimisation, again you could add try except statement when running your command to catch errors e.g.
https://stackoverflow.com/questions/24849998/how-to-catch-exception-output-from-python-subprocess-check-output



_Code snippet:_
```Python
     try:
         subprocess.run(isec_command, shell=True, check=True)
     except subprocess.CalledProcessError as e:
         print(e.output)
```
___
*[`URA-859/coverage_plots.py` line 60 at r3](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16#-OFvkM-GBtAvP6YMlF0h:-OFvkM-GBtAvP6YMlF0i:b-xv5ktx) ([raw file](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/coverage_plots.py#L60)):*
> ```Python
> 
>     # Raise an exception if no matching file is found
>     if file is None:
> ```

Could use try except to handle 'FileNotFoundError' exceptions
___
*[`URA-859/plots_bcftools.py` line 509 at r3](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16#-OFvopZ9E2xrgA_NRPvm:-OFvopZ9E2xrgA_NRPvn:b-prcij2) ([raw file](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L509)):*
> ```Python
> 
>         # Print out the sample name
>         print(sample)
> ```

Print statements can be removed since they were probably in the code during any debugging, these are scattered across some of your scripts 


<!-- Sent from Reviewable.io -->

Commit:
Address_comments: PEP8 and try-except statements

- Included try-except statements in coverage_plots.py and diagnostic_venn.py
- Addressed PEP8 issues.

Arun-bioinformatics commented on URA-859/plots_bcftools.py:
```diff
@@ -0,0 +1,667 @@
+        )
+        sp_oldpass_df_uniq = result[0]
+        sp_newpass_df_uniq = result[1]
+        sp_oldpass_df_common = result[2]
```
Done.
<!-- Reviewable comment -OGUqLd-9LRRkbSpnFUT:b-896fix -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Done.
<!-- Reviewable comment -OGUthzMCg9qyuj9dk_x:b-896fix -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Done.
<!-- Reviewable comment -OGUuvZ74uQrleFs3sBF:b-896fix -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Done.
<!-- Reviewable comment -OGZFIHME04EGn_8IE7T:b-896fix -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
Done.
<!-- Reviewable comment -OIPaduiAw1cPPKatmb1:b-896fix -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16)* status: all files reviewed, 4 unresolved discussions (waiting on @coderabbitai[bot] and @nadia31415)

___
*[`URA-859/coverage_plots.py` line 60 at r3](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16#-OFvkM-GBtAvP6YMlF0h:-OGUje4v0PZWSKTZQlKh:b-896fix) ([raw file](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/coverage_plots.py#L60)):*
<details><summary><i>Previously, nadia31415 wrote…</i></summary><blockquote>

Could use try except to handle 'FileNotFoundError' exceptions
</blockquote></details>

Done.
___
*[`URA-859/diagnostic_venn.py` line 96 at r3](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16#-OFvmqsq05mDkQSLzK08:-OGUsGXd9vgWLRqYPoBh:b-896fix) ([raw file](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/diagnostic_venn.py#L96)):*
<details><summary><i>Previously, nadia31415 wrote…</i></summary><blockquote>

More for optimisation, again you could add try except statement when running your command to catch errors e.g.
https://stackoverflow.com/questions/24849998/how-to-catch-exception-output-from-python-subprocess-check-output
</blockquote></details>

Done.
___
*[`URA-859/plots_bcftools.py` line 509 at r3](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16#-OFvopZ9E2xrgA_NRPvm:-OGUq2W86EmMSWjRdEFy:bb6un1j) ([raw file](https://github.com/eastgenomics/Haemonc_requests/blob/d2c8b8b53251ce4f65d038d8886c1c4d2c15be59/URA-859/plots_bcftools.py#L509)):*
<details><summary><i>Previously, nadia31415 wrote…</i></summary><blockquote>

Print statements can be removed since they were probably in the code during any debugging, these are scattered across some of your scripts
</blockquote></details>

Done


<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16)* status: 3 of 6 files reviewed, 4 unresolved discussions (waiting on @coderabbitai[bot] and @nadia31415)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
PEP-8 issues are addressed, but I am not sure if the comment is updated.

> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16#-:-OGUv-V-6H3O710rkXmF:br4eyyo)* status: 3 of 6 files reviewed, 4 unresolved discussions (waiting on @coderabbitai[bot] and @nadia31415)



<!-- Sent from Reviewable.io -->

nadia31415 reviewed (approved):
> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16)* status: all files reviewed, 1 unresolved discussion (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

nadia31415 reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16)* status: all files reviewed, 1 unresolved discussion (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

nadia31415 reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16)* status: all files reviewed, 1 unresolved discussion (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16)* status: all files reviewed, 1 unresolved discussion (waiting on @coderabbitai[bot])



<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/Haemonc_requests/16)* status: all files reviewed, 1 unresolved discussion (waiting on @coderabbitai[bot])



<!-- Sent from Reviewable.io -->
